src/main/java/com/labsynch/labseer/api/ApiSaltController.java

```diff
@@ -82,6 +95,191 @@ public ResponseEntity<String> showJson(@PathVariable("id") Long id) {
+					warnings.add(error);
+				}
+				List<Salt> saltsByAbbrev = Salt.findSaltsByAbbrevEquals(newSalt.getAbbrev()).getResultList();
+				if (saltsByAbbrev.size() > 1) {
+					logger.error("Number of salts found: " + saltsByAbbrev.size());
+					validSalt = false;
+					ErrorMessage error = new ErrorMessage();
+					error.setLevel("error");
+					error.setMessage("Duplicate salt abbreviation. Another salt exist with the same abbreviation.");
+					warnings.add(error);
+				}
+				List<Salt> saltsByFormula = Salt.findSaltsByFormulaEquals(newSalt.getFormula()).getResultList();
+				if (saltsByFormula.size() > 1) {
+					logger.error("Number of salts found: " + saltsByName.size());
+					validSalt = false;
+					ErrorMessage error = new ErrorMessage();
+					error.setLevel("error");
+					error.setMessage("Duplicate salt formula. Another salt exist with the same formula.");
+					warnings.add(error);
+				}
+				try { 
+					PurgeSaltDependencyCheckResponseDTO dependencyReport  = saltService.checkDependentData(oldSalt);
+					boolean hasDependencies = ! dependencyReport.isCanPurge();
+					if(hasDependencies){
+						ErrorMessage warning = new ErrorMessage();
+						warning.setLevel("warning");
+						warning.setMessage("This salt has experiment/lot dependencies. An attempt will be made to restandardize.");
+						warnings.add(warning);
+						// Need to Iterate Through Dependency Report and Individually Add As Warnings
+					}
+				}
+				catch (Exception e)
+				{
+					return new ResponseEntity<String>("ERROR: Depedency Check Issue:" + e.getMessage(), headers,
+						HttpStatus.INTERNAL_SERVER_ERROR);
+				}
+				if (validSalt && !dryrun) {
+					try {
+						oldSalt = saltStructureService.edit(oldSalt, newSalt);
+						try
+							{
+								// Attempt to Restandardize 
+								List<Long> saltID= new ArrayList<Long>();
+								saltID.add(oldSalt.getId());
+								standardizationService.restandardizeLots(saltID);
+
+
+								// Return Response of Any Warnings / Errors from Validations / Dependencies Done Prior 
+								return new ResponseEntity<String>(ErrorMessage.toJsonArray(warnings), headers, HttpStatus.OK);
+							}
+						catch (Exception e)
+							{
+								return new ResponseEntity<String>("ERROR: Restandardization Issue:" + e.getMessage(), headers,
+								HttpStatus.INTERNAL_SERVER_ERROR);
+							}
+					} catch (Exception e) {
+						logger.error("Error updating salt: " + e.getMessage());
+
+						validSalt = false;
+
+						ErrorMessage error = new ErrorMessage();
+						error.setLevel("error");
+						error.setMessage("Error updating salt: " + e.getMessage());
+						warnings.add(error);
+
+						return new ResponseEntity<String>(ErrorMessage.toJsonArray(warnings), headers, HttpStatus.BAD_REQUEST);
+					}
+				}
+				else if (!validSalt)
+				{
+					return new ResponseEntity<String>(ErrorMessage.toJsonArray(warnings), headers,
+					HttpStatus.EXPECTATION_FAILED);
```

Comment:
I don't think this is the right status code. This seems to have something to do with an "Expect" header: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/417

I think a BAD_REQUEST error would be more appropriate here

Reply:
Will replace this.